File: CSharpCodeBase/aiactions/snowtroll/ai_invite_to_hunt.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Select enemy using ai vision
local AI_InviteToHunt = Class({__includes=AINode, init=function(self, entity)
  AINode.init(self, entity)
end})

function AI_InviteToHunt:Visit()
  if self.status == NODE_READY then
    self.status = NODE_RUNNING
    self.entity:InviteToHunt()
  end
  if self.status == NODE_RUNNING then
    self.status = NODE_SUCCESS
  end
  return self.status
end

return AI_InviteToHunt

File: CSharpCodeBase/aiactions/snowtroll/ai_invite_to_hunt.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_invite_to_hunt :ainode{
 // Select enemy using ai vision
 var AI_InviteToHunt = Class({__includes=AINode, init=function(self, entity)
   AINode.init(self, entity);
 }})
 public void AI_InviteToHunt:Visit(){
   if(this.status == NODE_READY  ){ 
     this.status = NODE_RUNNING;
     this.entity:InviteToHunt();
   }
   if(this.status == NODE_RUNNING  ){ 
     this.status = NODE_SUCCESS;
   }
   return this.status;
 }
 return AI_InviteToHunt}}